src/filter_tree/filter_tree.c

```diff
@@ -187,11 +187,11 @@ FT_FilterNode *FilterTree_Combine
 	FT_FilterNode *root = NULL;
 
 	if(count > 0) {
-		root = filters[0];
+		root = FilterTree_Clone(filters[0]);
 		for(uint i = 1; i < count; i++) {
 			FT_FilterNode *and = FilterTree_CreateConditionFilter(OP_AND);
 			FilterTree_AppendLeftChild(and, root);
-			FilterTree_AppendRightChild(and, filters[i]);
+			FilterTree_AppendRightChild(and, FilterTree_Clone(filters[i]));
```

Comment:
I guess the logic is as follows:

when deconstructing a filter tree we want pointers to the original input
but when we constructing we want to clone ?

Reply:
it seems like the overall behavior isn't clear / consistent. we need to come up with some solid and clear rules here regarding memory ownership